lib/avro/builder/definition_cache.rb

```diff
@@ -48,20 +48,18 @@ def add_type_by_name(type_object, name, namespace = nil)
       # If the unqualified name is ambiguous then it is removed from the cache.
       # A set of unqualified names is kept to avoid reloading files for
       # ambiguous references.
-      def store_by_name(object, name = nil)
-        key = name || object.name.to_s
-        if schema_objects.key?(key)
-          schema_objects.delete(key)
-        elsif !schema_names.include?(key)
-          schema_objects.store(key, object)
+      def store_by_name(object, name = object.name.to_s)
+        if schema_objects.key?(name)
+          schema_objects.delete(name)
+        elsif !schema_names.include?(name)
```

Comment:
`exclude?` is from `activesupport` and I'd like to avoid pulling that in here if possible.